diff --git a/src/AdsConsent.ts b/src/AdsConsent.ts
index 0224ec5f94316c4b05b96939cc92ef7af5718f26..4f5cf920e4782eefa1239f616487b20cd0821c11 100644
--- a/src/AdsConsent.ts
+++ b/src/AdsConsent.ts
@@ -103,6 +103,10 @@ export const AdsConsent: AdsConsentInterface = {
   async getUserChoices(): Promise<AdsConsentUserChoices> {
     const tcString = await native.getTCString();
 
+	if (tcString.startsWith("DP1")) {
+      fixedTcString = tcString.replace(/^DP1/, "CP1");
+    } 
+
     let tcModel: TCModel;
 
     try {
